dataland-frontend/src/utils/DataTableDisplay.ts

```diff
@@ -43,3 +43,83 @@ export type ReportingPeriodOfDataSetWithId = {
   dataId: string;
   reportingPeriod: string;
 };
+
+/**
+ * Adds click event listeners on DataTable row headers to expand and collapse row
+ * @param expandedRowsOnClick function passes the latest list of expanded row id's
+ * @param newExpandedRowsCallback function that returns the updated (after click) list of expanded row id's
+ * @returns the map of rows and their click handlers needed for unmounting
+ */
+export function mountRowHeaderClickEventListeners(
+  expandedRowsOnClick: () => string[],
+  newExpandedRowsCallback: (newExpandedRows: string[]) => void,
+): Map<Element, EventListener> {
+  const handlerMap: Map<Element | HTMLButtonElement, EventListener> = new Map();
+  let expandedRowGroups: string[] = [];
+
+  const t = setTimeout(() => {
+    const buttonAttributeName = "data-parent-id";
+    const rows = Array.from(document.querySelectorAll("[data-row-header-click]")).filter((el) => el);
+    const rowButtons = rows
+      .map((el: Element) => {
+        const button = el.parentNode?.querySelector('button[data-pc-section="rowgrouptoggler"]');
+        if (button) {
+          button?.setAttribute(buttonAttributeName, el.id);
+          return button;
+        }
+        return void 0;
```

Comment:
Why is this "return void" needed?  The callback of a `.map()` should be able to handle it, if `undefined` is returned.

Reply:
The `.map()` receives an element which is a child of our target, so it isn't (and doesn't return) `undefined`. We do a check in the parent of that element for the button we need, if we don't then we forcefully set it to undefined... then filter them out so we only get a list of the buttons we need. We only want to do operations on buttons which contain the appropriate attribute.

Reply:
Ok then let's keep it.  The way I see it, the case that a button is not found is kind of a "fallback" and shouldn't happen anyways.